agent/grpc-external/services/peerstream/stream_resources.go

```diff
@@ -215,7 +219,10 @@ func (s *Server) HandleStream(streamReq HandleStreamRequest) error {
 		})
 		logTraceSend(logger, sub)
 
-		if err := streamReq.Stream.Send(sub); err != nil {
+		sendMutex.Lock()
```

Comment:
Do you think it would be worthwhile create a func like:

```
sendLocked := func(msg *pbpeerstream.ReplicationMessage) error {
   sendMutex.Lock()
   defer sendMutext.Unlock()
   return streamReq.Stream.Send(msg)
}
```

And then use this everywhere as opposed to copying the mutex handling a bunch of places.

Reply:
Yep, I like that.

Reply:
done